src/d3a/models/market/two_sided_pay_as_bid.py

```diff
@@ -206,13 +206,15 @@ def _perform_pay_as_bid_matching(self):
         return offer_bid_pairs
 
     def accept_bid_offer_pair(self, bid, offer, clearing_rate, trade_bid_info, selected_energy):
+        already_tracked = bid.buyer == offer.seller
         trade = self.accept_offer(offer_or_id=offer,
                                   buyer=bid.buyer,
                                   energy=selected_energy,
                                   trade_rate=clearing_rate,
-                                  already_tracked=False,
+                                  already_tracked=already_tracked,
```

Comment:
Why is it needed here ? The reason I am asking is because if `already_tracked` is true, this will lead to tracking the same trade twice.

Reply:
This is needed for the case dispatching bottom-> up and offers and bid matching doesnt happen in direct line fo connection. Meaning in the default_2a case if the Cell Tower bid gets matched with the CEP offer in  House 1 (and not in grid or House 2, as expected).

Reply:
There is a Trade from IAA House 1 --> IAA House 1 without this change.

Reply:
An offline discussion revealed that this change indeed makes sense. The trade from and to House 1 market is not tracked with this change. 
@faizan2590 If you need more information, please ping me and we can discuss offline.

Reply:
Yes, please. Let me know when you are free